python/ray/autoscaler/autoscaler.py

```diff
@@ -140,13 +141,16 @@ def _update(self):
             return
 
         self.last_update_time = now
-        nodes = self.workers()
+        all_nodes = self.workers()
+        nodes = self.managed_workers()
         # Check pending nodes immediately after fetching the number of running
         # nodes to minimize chance number of pending nodes changing after
-        # additional nodes are launched.
+        # additional all_nodes are launched.
         num_pending = self.pending_launches.value
         self.load_metrics.prune_active_ips(
-            [self.provider.internal_ip(node_id) for node_id in nodes])
+            [self.provider.internal_ip(node_id) for node_id in all_nodes])
```

Comment:
```suggestion
            [self.provider.internal_ip(node_id) for node_id in self.all_workers()])
```

Why not avoid making this temporary variable entirely like this?

Reply:
+1